### PR TITLE
chore(IDX): enable RUN_ON_DIFF_ONLY on bazel-test-darwin-x86-64

### PR DESCRIPTION
**Pull request by basvandijk**
We used to test many targets (`//rs/...` and `//publish/binaries/...`) on darwin on PRs even if no dependencies of them got changed. We now enable RUN_ON_DIFF like we do for `bazel-test-all` to only test the targets that have changed dependencies. This will reduce the current P95 of almost 90 minutes (current P50 is over an 1 hour, P5 is 42 min).

### CODE REVIEW

**Comment by cgundy** [comment]
Hey is this ready for review? We had discussed some more things in our slack thread regarding this change. Should we continue the discussion there or would you prefer here on this PR?

File: gitlab-ci/src/bazel-ci/diff.sh [hunk 1]
```diff
@@ -52,10 +52,11 @@ if [ ${#files[@]} -eq 0 ]; then
     exit 0
 fi
 
+UNIVERSE="$(echo ${BAZEL_TARGETS:-//...} | sed 's/ /+/')"
```
Comment (nmattia) [inline gitlab-ci/src/bazel-ci/diff.sh:55]: What's `UNIVERSE` and why the substitution?

Comment (basvandijk) [reply gitlab-ci/src/bazel-ci/diff.sh:60]: Good question. I added some docs.

**Review by nmattia — APPROVED** [review]
Thanks! out of curiosity, will this trigger a rebuild on e.g. a `WORKSPACE` change?

**Comment by basvandijk** [comment]
> Thanks! out of curiosity, will this trigger a rebuild on e.g. a `WORKSPACE` change?

Yes. [Any change to a bazel file will run all bazel targets](https://sourcegraph.com/github.com/dfinity/ic/-/blob/gitlab-ci/src/bazel-ci/diff.sh?L21-L25).

**Review by nmattia — CHANGES_REQUESTED** [review]
@basvandijk I just talked to Carly and realized I don't actually understand what this PR does :) Also, maybe we should focus our efforts on the namespace runners for the time being?

**Review by nmattia — APPROVED** [review]
Rubber stamp

**Review by cgundy — APPROVED** [review]

File: rs/pocket_ic_server/src/main.rs [hunk 2]
```diff
@@ -78,6 +78,7 @@ fn current_binary_path() -> Option<PathBuf> {
     std::env::args().next().map(PathBuf::from)
 }
 
+/// Some bogus comment to trigger a build
```
Comment (mraszyk) [inline rs/pocket_ic_server/src/main.rs:81]: Are you going to remove the comment again?

Comment (basvandijk) [reply rs/pocket_ic_server/src/main.rs:81]: Yes, once it works. I turned the PR back to draft since it needs more work.

**Pull request closed without merging**